rmw_connext_dynamic_cpp/CMakeLists.txt

```diff
@@ -17,20 +17,23 @@ if(NOT Connext_FOUND)
 endif()
 
 find_package(rmw REQUIRED)
+find_package(rmw_connext_shared_cpp REQUIRED)
 find_package(rosidl_generator_cpp REQUIRED)
 
 find_package(rosidl_typesupport_introspection_cpp REQUIRED)
+
 ament_export_dependencies(rosidl_typesupport_introspection_cpp)
 ament_export_dependencies(rmw rosidl_generator_cpp)
-ament_export_include_directories(${Connext_INCLUDE_DIRS})
+ament_export_include_directories(${rmw_connext_shared_cpp_INCLUDE_DIRS})
 
 register_rmw_implementation("rosidl_typesupport_introspection_cpp")
 
-link_directories("${Connext_LIBRARY_DIRS}")
+link_directories("${rmw_connext_shared_cpp_LIBRARIES}")
```

Comment:
This is wrong usage of the function. The arguments to `link_directories` are one or more directories (http://www.cmake.org/cmake/help/v2.8.12/cmake.html#command:link_directories) but this is passing the libraries to the function.

As mentioned before calling `link_directories()` is not necessary at all.